Introduction of new spacing grid (#3285)
ONSdigital/design-system: migration_guides/70.x.x-to-71.0.0-migration-guide.md

-   **Components impacted:**

    -   Any component or layout where you have manually set our margin or padding utility classes.

-   **Migration steps:**

    1. Locate all instances of the utility classes in your codebase.
    2. Replace old size name with new corresponding size name:

    -   <details>
        <summary><b>Click for example:</b></summary>

        ```scss
        //OLD Sizes
        $sizes: (
            no: 0,
            xxs: 0.25rem,
            xs: 0.5rem,
            s: 1rem,
            m: 1.5rem,
            l: 2rem,
            xl: 3rem,
            auto: auto,
        );

        //NEW Sizes
        $sizes: (
            no: 0,
            4xs: 0.125rem,
            3xs: 0.25rem,
            2xs: 0.5rem,
            xs: 0.75rem,
            s: 1rem,
            m: 1.25rem,
            l: 1.5rem,
            xl: 2rem,
            2xl: 2.5rem,
            3xl: 3rem,
            4xl: 4rem,
            5xl: 5rem,
            auto: auto,
        );
        ```

        </details>

### **_Update icon sizes naming standard_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

> -   **Description of change:** Icon sizes have been updated with the new naming size standard.

Reviewer: Think this needs a bit more information like what has been done for the padding above

-   **Reason for change:** Updated icon sizes to match the naming size standard.

    This will be a breaking change and will require the renaming of some parameters:

-   **Components impacted:**

    Any component or layout where you have manually set the icon size parameter.

-   **Migration steps:**

    1. Locate all instances of the `iconSize` parameter in your codebase.
    2. Replace old size name with new corresponding size name:

    -   <details>
        <summary><b>Click for example:</b></summary>

        ```njk
        //OLD Sizes
        $sizes:
        [
            "iconSize": 'xxxl',
            "iconSize": 'xxl'
            "iconSize": 'xl'
            "iconSize": 'l'
            "iconSize": 'm'
            "iconSize": 's'
        ];

        //NEW Sizes
        $sizes:
        [
            "iconSize": '3xl',
            "iconSize": '2xl'
            "iconSize": 'xl'
            "iconSize": 'l'
            "iconSize": 'm'
            "iconSize": 's'
        ];
        ```

        </details>

### **_Update font sizes naming standard_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

> -   **Description of change:** font sizes have been updated with the new naming size standard.

Reviewer: Same here a bit more information

-   **Reason for change:** Updated font sizes to match the naming size standard.

-   **Components impacted:**

    Any component or layout where you have manually set the font-size utility classes.

-   **Migration steps:**

    1. Locate all instances of the font size utility classes in your codebase.
    2. Replace old size name with new corresponding size name:

    -   <details>
        <summary><b>Click for example:</b></summary>

        ```
        //OLD Sizes
        $sizes:
        [
            'ons-u-fs-xxxl',
            'ons-u-fs-xxl',
            'ons-u-fs-xl',
            'ons-u-fs-l',
            'ons-u-fs-m',
            'ons-u-fs-r--b',
            'ons-u-fs-r',
            'ons-u-fs-s--b',
            'ons-u-fs-s'
        ];

        //NEW Sizes
        $sizes:
        [
            'ons-u-fs-3xl',
            'ons-u-fs-2xl',
            'ons-u-fs-xl',
            'ons-u-fs-l',
            'ons-u-fs-m',
            'ons-u-fs-r--b',
            'ons-u-fs-r',
            'ons-u-fs-s--b',

>              'ons-u-fs-s'

Reviewer: Needs a bit of aligning here

-   **Components impacted:**

    -   Any component or layout where you have manually set our margin or padding utility classes.

-   **Migration steps:**

    1. Locate all instances of the utility classes in your codebase.
    2. Replace old size name with new corresponding size name:

    -   <details>
        <summary><b>Click for example:</b></summary>

        ```scss
        //OLD Sizes
        $sizes: (
            no: 0,
            xxs: 0.25rem,
            xs: 0.5rem,
            s: 1rem,
            m: 1.5rem,
            l: 2rem,
            xl: 3rem,
            auto: auto,
        );

        //NEW Sizes
        $sizes: (
            no: 0,
            4xs: 0.125rem,
            3xs: 0.25rem,
            2xs: 0.5rem,
            xs: 0.75rem,
            s: 1rem,
            m: 1.25rem,
            l: 1.5rem,
            xl: 2rem,
            2xl: 2.5rem,
            3xl: 3rem,
            4xl: 4rem,
            5xl: 5rem,
            auto: auto,
        );
        ```

        </details>

### **_Standardisation of icon size naming_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

> -   **Description of change:** Icon sizes have been updated with to new size naming standard. This change replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), similar to the one used for padding and margins.

Reviewer: ```suggestion
-   **Description of change:** Icon sizes have been updated to reflect the new size naming convention. This change replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), the same as the one that has been brought for padding and margins.
```

-   **Reason for change:** This update standardises how sizes are referenced across the system.

-   **Components impacted:**

    Anywhere an icon size parameter is set to xxxl or xxl.

-   **Migration steps:**

    1. Locate all instances of the `iconSize` parameter in your codebase.
    2. Replace old size name with new corresponding size name:

    -   <details>
        <summary><b>Click for example:</b></summary>

        ```njk
        //OLD Sizes
        $sizes:
        [
            "iconSize": 'xxxl',
            "iconSize": 'xxl'
            "iconSize": 'xl'
            "iconSize": 'l'
            "iconSize": 'm'
            "iconSize": 's'
        ];

        //NEW Sizes
        $sizes:
        [
            "iconSize": '3xl',
            "iconSize": '2xl'
            "iconSize": 'xl'
            "iconSize": 'l'
            "iconSize": 'm'
            "iconSize": 's'
        ];
        ```

        </details>

### **_Standardisation of font size naming_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

> -   **Description of change:** Font size classes have been updated to reflect the new size naming convention in the Design System. This update replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), similar to the one used for padding and margins.

Reviewer: ```suggestion
-   **Description of change:** Font size classes have been updated to reflect the new size naming convention. This change replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), the same as the one that has been brought for padding and margins.
```

> -   **Reason for change:** This update standardises how sizes are referenced across the system

Reviewer: ```suggestion
-   **Reason for change:** This update standardises how sizes are referenced across the system.
```

-   **Components impacted:**

    Anywhere you use the xxl or xxxl font-size utility classes.

-   **Migration steps:**

    1. Locate all instances of the font size utility classes in your codebase.
    2. Replace old size name with new corresponding size name:

    -   <details>
        <summary><b>Click for example:</b></summary>

        ```
        //OLD Sizes
        $sizes:
        [
            'ons-u-fs-xxxl',
            'ons-u-fs-xxl',
            'ons-u-fs-xl',
            'ons-u-fs-l',
            'ons-u-fs-m',
            'ons-u-fs-r--b',
            'ons-u-fs-r',
            'ons-u-fs-s--b',
            'ons-u-fs-s'
        ];

        //NEW Sizes
        $sizes:
        [
            'ons-u-fs-3xl',
            'ons-u-fs-2xl',
            'ons-u-fs-xl',
            'ons-u-fs-l',
            'ons-u-fs-m',
            'ons-u-fs-r--b',
            'ons-u-fs-r',
            'ons-u-fs-s--b',
            'ons-u-fs-s'
        ];
        ```

        </details>

### **_Standardisation of breakpoint size naming_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

> -   **Description of change:** The breakpoint naming convention has been updated to match other size naming like the padding and margin sizes. The old breakpoint size names like xxs and xxl have been replaced with 2xs, 2xl, and so on, to better align with other size naming.

Reviewer: ```suggestion
-   **Description of change:** Breakpoint size names have been updated to reflect the new size naming convention. This change replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), the same as the one that has been brought for padding and margins.
```

        </details>

> ### **_Update to icon sizes naming standard_**

Reviewer: I think the wording of this is a bit more clear

### **_Standardisation of breakpoint size naming_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

-   **Description of change:** The breakpoint naming convention has been updated to match other size naming like the padding and margin sizes. The old breakpoint size names like xxs and xxl have been replaced with 2xs, 2xl, and so on, to better align with other size naming.

-   **Reason for change:** The new naming convention standardises how breakpoints are referenced across the design system.

-   **Components impacted:**

    Any component or layout that references breakpoints using the previous naming convention will need to be updated.

-   **Migration steps:**

    1. Locate all instances of breakpoints in your codebase.
    2. Replace old breakpoint names with the new corresponding names:

> ### **_Update to icon sizes naming standard_**

Reviewer: I think this is worded a bit better

### **_Standardisation of font size naming_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

-   **Description of change:** Font size classes have been updated to reflect the new size naming convention in the Design System. This update replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), similar to the one used for padding and margins.

-   **Reason for change:** This update standardises how sizes are referenced across the system

-   **Components impacted:**

    Any component or layout where you use the xxl or xxxl font-size utility classes.

-   **Migration steps:**

    1. Locate all instances of the font size utility classes in your codebase.
    2. Replace old size name with new corresponding size name:

        </details>

### **_Update to font sizes naming standard_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

-   **Description of change:** Font size classes have been updated to reflect the new naming conventions in the Design System. This update replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), similar to the one used for spacing and other components.
-   **Reason for change:** This update standardizes how sizes are referenced across the system

> -   **Components impacted:**

Reviewer: I think this is worded a bit better and is more consistent with others

### **_Standardisation of font size naming_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

-   **Description of change:** Font size classes have been updated to reflect the new size naming convention in the Design System. This update replaces the old naming convention (e.g., xxxl, xxl) with a numerical system (e.g., 3xl, 2xl), similar to the one used for padding and margins.

-   **Reason for change:** This update standardises how sizes are referenced across the system

-   **Components impacted:**

    Anywhere you use the xxl or xxxl font-size utility classes.

-   **Migration steps:**

    1. Locate all instances of the font size utility classes in your codebase.
    2. Replace old size name with new corresponding size name:

        </details>

### **_Update to Grid Breakpoint Naming Standard_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

-   **Description of change:** The grid breakpoint naming convention has been updated to a more consistent numerical system. The old breakpoint names like xxs and xxl have been replaced with 2xs, 2xl, and so on, to better align with the numeric sizing system.

-   **Reason for change:** The new naming convention standardizes how breakpoints are referenced across the design system.

-   **Components impacted:**

    Any component or layout that references grid breakpoints using the previous naming conventions will need to be updated

-   **Migration steps:**

    1. Locate all instances of grid breakpoints in your codebase.

>     2. Replace old breakpoint names with the new corresponding names:

Reviewer: I think the wording of this is a bit more clear

### **_Standardisation of breakpoint size naming_**

[**Link to PR**](https://github.com/ONSdigital/design-system/pull/3285)

-   **Description of change:** The breakpoint naming convention has been updated to match other size naming like the padding and margin sizes. The old breakpoint size names like xxs and xxl have been replaced with 2xs, 2xl, and so on, to better align with other size naming.

-   **Reason for change:** The new naming convention standardises how breakpoints are referenced across the design system.

-   **Components impacted:**

    Any component or layout that references breakpoints using the previous naming convention will need to be updated.

-   **Migration steps:**

    1. Locate all instances of breakpoints in your codebase.
    2. Replace old breakpoint names with the new corresponding names:
